### PR TITLE
Remove the util crate

### PR DESCRIPTION
**Pull request by asajeffrey**
This PR removes the `util` crate.

* Replaced the `spawn_named` and `clamp` functions by appropriate uses of `std::thread::Builder::spawn`, `std::cmp::min` and `std::cmp::max`.
* Moved `opts`, `prefs` and `resource_files` into a new `config` crate.
* Moved `remutex` and `geometry` into their own crates.

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes do not require tests because they are refactorings

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/14592)

### CODE REVIEW

File: components/geometry/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+[package]
```
Comment (mbrubeck) [inline components/geometry/Cargo.toml]: Nit: This addition is in the wrong commit.

File: components/remutex/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+[package]
```
Comment (mbrubeck) [inline components/remutex/Cargo.toml]: Nit: This addition is in the wrong commit.

File: tests/unit/remutex/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+[package]
```
Comment (mbrubeck) [inline tests/unit/remutex/Cargo.toml]: Nit: This addition is in the wrong commit.

File: tests/unit/util/thread.rs [hunk 4]
```diff
@@ -1,13 +0,0 @@
-/* This Source Code Form is subject to the terms of the Mozilla Public
```
Comment (mbrubeck) [inline tests/unit/util/thread.rs]: This deletion is in the wrong commit.

**Review by mbrubeck — CHANGES_REQUESTED** [review]
r=mbrubeck with some nits. (Could be fixed by just squashing into a single commit.)

**Comment by asajeffrey** [comment]
Squashed. @bors-servo: r=mbrubeck

**Comment by KiChjang** [comment]
Unit tests imports need to be updated, I presume?

**Comment by asajeffrey** [comment]
@KiChjang indeed, doing that right now. Ditto cef.

**Comment by asajeffrey** [comment]
Updated unit tests and cef port. @bors-servo r=mbrubreck p=10

**Comment by KiChjang** [comment]
```
error: unused import
  --> /home/servo/buildbot/slave/linux-dev/build/target/geckolib/debug/build/style-584c140631e9c02b/out/properties.rs:31:5
   |
31 | use servo_config::prefs::PREFS;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^
/home/servo/buildbot/slave/linux-dev/build/components/style/lib.rs:155:5: 155:58 note: in this expansion of include!
   |
note: lint level defined here
  --> /home/servo/buildbot/slave/linux-dev/build/components/style/lib.rs:30:9
   |
30 | #![deny(warnings)]
   |         ^^^^^^^^
```

**Comment by asajeffrey** [comment]
Oh bah humbug, geckolib.

**Comment by asajeffrey** [comment]
Fixed geckolib @bors-servo r=mbrubeck

**Pull request merged**